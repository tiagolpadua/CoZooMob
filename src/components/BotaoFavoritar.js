import {Icon} from 'native-base';
import React, {Component} from 'react';
import {TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';

export default class BotaoFavoritar extends Component {
  static propTypes = {
    favoritado: PropTypes.bool.isRequired,
    favoritarCallback: PropTypes.func.isRequired,
    desfavoritarCallback: PropTypes.func.isRequired,
  };

  render() {
    const {favoritado, favoritarCallback, desfavoritarCallback} = this.props;

    if (favoritado) {
      return (
        <TouchableOpacity onPress={desfavoritarCallback}>
          <Icon name="star" />
        </TouchableOpacity>
      );
    } else {
      return (
        <TouchableOpacity onPress={favoritarCallback}>
          <Icon name="star-outline" />
        </TouchableOpacity>
      );
    }
  }
}
