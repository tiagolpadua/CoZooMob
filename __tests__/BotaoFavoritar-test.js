/**
 * @format
 */

import 'react-native';
import React from 'react';
import BotaoFavoritar from '../src/components/BotaoFavoritar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renderiza se favoritado é true', () => {
  const tree = renderer
    .create(
      <BotaoFavoritar
        favoritado={true}
        favoritarCallback={() => {}}
        desfavoritarCallback={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renderiza se favoritado é false', () => {
  const tree = renderer
    .create(
      <BotaoFavoritar
        favoritado={false}
        favoritarCallback={() => {}}
        desfavoritarCallback={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
