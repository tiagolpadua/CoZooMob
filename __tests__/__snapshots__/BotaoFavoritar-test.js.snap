// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza se favoritado é false 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#000",
            "fontSize": 30,
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renderiza se favoritado é true 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#000",
            "fontSize": 30,
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
